server/message_hooks.go

```diff
@@ -670,8 +724,13 @@ func (p *Plugin) Delete(teamID, channelID string, user *model.User, post *model.
 	return nil
 }
 
-func (p *Plugin) DeleteChat(chatID string, user *model.User, post *model.Post) error {
-	client, err := p.GetClientForUser(user.Id)
+func (p *Plugin) DeleteChat(userID string, post *model.Post) error {
```

Comment:
Are we expecting that the user ID will be other than the post author? If not, I think with only with the post as argument should be enough

Reply:
Good question: @jespino, do you know if there are cases where we `DeleteChat` for a different user?

Reply:
A system admin can delete other user post, but not only that, in theory, the permissions system allows that anybody that has permission to delete others posts in a channel, can deleted that posts, and that can be configured with the advanced permissions system.

Reply:
In those cases, would the `userID` passed as a parameter be different from `post.UserId`? It seems redundant at present, so I'm guessing even for the system admin cases we could just pass `post`.

Reply:
Yes, maybe it is possible that we can use the post.UserId, because we are using it to decide in behave of who we delete the post. I think it is safe enough to use the post.UserId directly.